### ISSUE TITLE
JSON parse error

### ISSUE THREAD

**Issue by pyjsorg** [pyjs/original-pyjs#720]
dynamictable example (some of my stuff too) doesn't work because of a JSON parsing error. This worked in 0.7 and still works in pyjd hulahop.

See example json; resulting error is &quot;ReferenceError: TimeSlot is not defined&quot;

Git bisect indicates 503d55e62d0434f0b14856f9cfa848f5a2830316 is first bad commit.

This is unexpected because that is a fix to PanelBase - seems strange. Confirmed by commenting out def append() through to def **len**(), symptoms then disappear.

_Original issue: http://code.google.com/p/pyjamas/issues/detail?id=581 (April 16, 2011 01:14:10)_

**Comment by pyjsorg** [comment]
_From cornelis...@gmail.com on April 16, 2011 07:28:38:_
Nice find. Will definitely lead us to a solution.

**Comment by pyjsorg** [comment]
_From jameskhe...@gmail.com on April 16, 2011 23:51:51:_
Ah, forget the whole thing; this seems to have randomly started working. Definitely wasn't yesterday... I guess that's good news! :S

**Comment by pyjsorg** [comment]
_From cornelis...@gmail.com on April 18, 2011 09:28:47:_
No don't :-)

Fix to get going after 503d55e62d0434f0b14856f9cfa848f5a2830316 is to change
'if slot:' to 'if slot is not None:' in DynaTable.py

Then I got stuck at commit 849093abb4fc0478ce7ad68db829f0897f894ff7
Had to fix ui/RootPanel.py 'if Element is None:' to 'if Element is not None:',
which was also fixed in acf6ac5aaef8191765cd463feed3767ab7255d4f.

Finally I got at bad commit 0fd32a25edeae18246258df4ddc3a4bce72897de, which did make sense to me as a possible source for **jsonclass** bug. From that point on, the classes are not defined in the global javascript namespace. This lead to the fix in 4ee1c452babc21f19c4e767dfd7cb608d043719d

I'm closing this issue as fixed.
